packages/protocol/contracts/governance/Election.sol

```diff
@@ -0,0 +1,779 @@
+   */
+  function getEligibleValidatorGroupsVoteTotals()
+    external
+    view
+    returns (address[] memory, uint256[] memory)
+  {
+    return votes.total.eligible.getElements();
+  }
+
+  function validatorAddressFromCurrentSet(uint256 index) external view returns (address) {
+    address validatorAddress;
+    assembly {
+      let newCallDataPosition := mload(0x40)
+      mstore(newCallDataPosition, index)
+      let success := staticcall(5000, 0xfa, newCallDataPosition, 32, 0, 0)
+      returndatacopy(add(newCallDataPosition, 64), 0, 32)
+      validatorAddress := mload(add(newCallDataPosition, 64))
+    }
+
+    return validatorAddress;
+  }
+
+  function numberValidatorsInCurrentSet() external view returns (uint256) {
+    uint256 numberValidators;
+    assembly {
+      let success := staticcall(5000, 0xf9, 0, 0, 0, 0)
+      let returnData := mload(0x40)
+      returndatacopy(returnData, 0, 32)
+      numberValidators := mload(returnData)
+    }
+
+    return numberValidators;
+  }
+
+  /**
+   * @notice Returns a list of elected validators with seats allocated to groups via the D'Hondt
+   *   method.
+   * @return The list of elected validators.
+   * @dev See https://en.wikipedia.org/wiki/D%27Hondt_method#Allocation for more information.
+   */
+  function electValidators() external view returns (address[] memory) {
+    // Only members of these validator groups are eligible for election.
+    uint256 maxNumElectionGroups = Math.min(
+      maxElectableValidators,
+      votes.total.eligible.list.numElements
+    );
+    /*
+    uint256 requiredVotes = electabilityThreshold.multiply(
+      FixidityLib.newFixed(votes.total.total)
+    ).fromFixed();
+    */
+    // TODO(asa): Filter by > requiredVotes
+    address[] memory electionGroups = votes.total.eligible.headN(maxNumElectionGroups);
+    uint256[] memory numMembers = getValidators().getGroupsNumMembers(electionGroups);
+    // Holds the number of members elected for each of the eligible validator groups.
+    uint256[] memory numMembersElected = new uint256[](electionGroups.length);
+    uint256 totalNumMembersElected = 0;
+    // Assign a number of seats to each validator group.
+    while (totalNumMembersElected < maxElectableValidators) {
+      uint256 groupIndex = 0;
+      bool memberElected = false;
+      (groupIndex, memberElected) = dHondt(electionGroups, numMembers, numMembersElected);
+
+      if (memberElected) {
+        numMembersElected[groupIndex] = numMembersElected[groupIndex].add(1);
+        totalNumMembersElected = totalNumMembersElected.add(1);
+      } else {
+        break;
+      }
+    }
+    require(totalNumMembersElected >= minElectableValidators);
+    // Grab the top validators from each group that won seats.
+    address[] memory electedValidators = new address[](totalNumMembersElected);
+    totalNumMembersElected = 0;
+    for (uint256 i = 0; i < electionGroups.length; i = i.add(1)) {
```

Comment:
no need for safe math

Reply:
I don't feel very strongly about it... either way works
I remember reading [Counter](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/drafts/Counters.sol) docs which state:
```
 Since it is not possible to overflow a 256 bit integer with increments of one, `increment` can skip the {SafeMath}
 overflow check, thereby saving gas. This does assume however correct usage, in that the underlying `_value` is never
 directly accessed.
 ```
 which sounds reasonable.